functions/src/middlewares/rateLimiter.ts

```diff
@@ -0,0 +1,33 @@
+import { NextFunction, Request, Response } from 'express';
+import * as Redis from 'ioredis';
+import {
+  IRateLimiterStoreOptions,
+  RateLimiterRedis,
+} from 'rate-limiter-flexible';
+
+const redis = new Redis(process.env.REDIS_URI); // uses defaults unless given configuration object
+
+const options: IRateLimiterStoreOptions = {
+  storeClient: redis,
+  keyPrefix: `middleware-${process.env.ENV ?? 'default'}`,
+  points: parseInt(process.env.RATE_LIMIT_POINTS ?? '10'), // 10 requests
+  duration: 1, // per 1 second by IP
+};
+
+const rateLimiter = new RateLimiterRedis(options);
+
+export const rateLimiterMiddleware = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
any way this can be typed?

Reply:
a think a `void` might be fine looking back at this.